doc/building.rst

```diff
@@ -15,88 +15,151 @@ These instructions are updated to work with version |version| of the Nextcloud C
 
 When you no part of the nextcloud organisation, clone with HTTPS:
 
-```
-$ git clone git@github.com:nextcloud/desktop.git
-```
+.. code-block:: bash
+
+   $ git clone git@github.com:nextcloud/desktop.git
+
 
 macOS
 -----
 
-In addition to needing XCode (along with the command line tools), developing in
-the macOS environment requires extra dependencies.  You can install these
-dependencies through MacPorts_ or Homebrew_.  These dependencies are required
-only on the build machine, because non-standard libs are deployed in the app
-bundle.
+.. note:: While it is possible to do many of the following steps using GUI
+   frontends, wherever possible the Terminal commands are listed instead, in order
+   to streamline the process.
+
+1. Install Xcode from the Mac App Store:
+
+   https://apps.apple.com/app/xcode/id497799835
+
+Then, in Terminal:
+
+2. Install Xcode command line tools:
+
+   .. code-block:: bash
+
+      % xcode-select –install
+
+3. Install Homebrew from `brew.sh`_ (which will just give you the following):
+
+.. _`brew.sh`: https://brew.sh
+
+   .. code-block:: bash
+
+      % /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+
+4. Fix Homebrew permissions. (This is `a known issue`_, and, yes, the workaround may lead to other shell warnings.)
+
+.. _`a known issue`: https://stackoverflow.com/a/63241724
+
+   .. code-block:: bash
+
+      % sudo chown -R $(whoami):admin /usr/local/* \
+         && sudo chmod -R g+rwx /usr/local/*
+
+5. Install Homebrew packages:
+
+   .. code-block:: bash
+
+      % brew install git qt qt5 qtkeychain cmake openssl pkgconfig glib doxygen cmocka
+
+6. Certain Homebrew packages are not automatically linked in places where
+   the build scripts can find them, so add them to your shell profile:
 
-The tested and preferred way to develop in this environment is through the use
-of HomeBrew_. The team has its own repository containing non-standard
-recipes.
+   .. code-block:: bash
 
-To set up your build environment for development using HomeBrew_:
+      % echo 'export OPENSSL_ROOT_DIR="/usr/local/opt/openssl@1.1"' >> ~/.zshrc
+      % echo 'export Qt5_DIR="/usr/local/opt/qt"' >> ~/.zshrc
+      % echo 'export Qt5LinguistTools_DIR="/usr/local/opt/qt/lib/cmake/Qt5LinguistTools/"' >> ~/.zshrc
 
-1. Install Xcode
-2. Install Xcode command line tools::
-    xcode-select --install
+   .. note::
 
-3. Install homebrew::
-    /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+      These commands assume you are using ``zsh`` instead of ``bash``, `as recommended by Apple`_.
+      Typically, if you are using ``zsh`` (the norm on macOS), your prompt will be a ``%``,
+      whereas if you are using ``bash`` (the norm on Linux), your prompt will be a ``$``.
+      If you prefer to use ``bash`` instead of ``zsh``, replace ``~/.zshrc`` with ``~/.bashrc``.
```

Comment:
I would drop the explanation about the appearance of the prompt. It's misleading. Everyone can change it and does so. If you want to find out if you are running bash or zsh you can check if the variable BASH_VERSION or ZSH_VERSION is set. If BASH_VERSION is set, you are running on bash. If the ZSH_VERSION variable is set, you running on zsh. But see my other comment what I think about `.zshrc` :)

Reply:
The important thing is if the user needs to know what shell they're using, the instructions should specify a way of finding out. While in your personal circle it may be the norm for people to change the shell prompt, the sort of person who doesn't know what shell they're using presumably hasn't.

Reply:
The user just needs to enter `echo $BASH_VERSION` and press enter. If this gives an output, you are using bash. If not, enter `echo $ZSH_VERSION`. If this gives you an output you are running zsh. If both commands give no output you are most likely running a different shell. I think this method is more reliable because one day the developers of zsh change the default prompt and then the instructions become out of date (the developers of zsh could also decide to drop ZSH_VERSION but I would consider this more unlikely).

Reply:
Of course, moving the environmental variables into a non-shell-specific source file eliminates the need to make a distinction between `bash` and `zsh` in the first.

Reply:
Could we change the text to what @FlexW suggested?